pkg/controller/jobs/job/job_controller.go

```diff
@@ -237,7 +253,16 @@ func (j *Job) EquivalentToWorkload(wl kueue.Workload) bool {
 		return false
 	}
 
-	if *j.Spec.Parallelism != wl.Spec.PodSets[0].Count {
+	ps0 := &wl.Spec.PodSets[0]
+	if mpc := j.minPodsCount(); mpc != nil {
```

Comment:
This is a bit hard to follow.
Can you extract it into a separate function `compareNullable`? You could probably implement it using generics.